libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -529,55 +692,94 @@ void AP_SerialManager::init()
                 case SerialProtocol_DJI_FPV:
                 case SerialProtocol_MSP_DisplayPort:
                     // baudrate defaults to 115200
-                    state[i].baud.set_default(AP_SERIALMANAGER_MSP_BAUD/1000);
-                    uart->begin(state[i].baudrate(),
+                    serial_state(i).set_and_default_baud(AP_SERIALMANAGER_MSP_BAUD/1000);
+                    uart->begin(serial_state(i).baudrate(),
                                          AP_SERIALMANAGER_MSP_BUFSIZE_RX,
                                          AP_SERIALMANAGER_MSP_BUFSIZE_TX);
                     uart->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
                     // Note init is handled by AP_MSP
                     break;
 #endif
                 default:
-                    uart->begin(state[i].baudrate());
+                    uart->begin(serial_state(i).baudrate());
             }
         }
     }
 }
 
+AP_SerialManager::SerialState& AP_SerialManager::serial_state(uint8_t i)
```

Comment:
`i` is const, could change function signature

Reply:
Yup. I blame @bugobliterator :)